src/clims/services/substance.py

```diff
@@ -25,32 +25,40 @@ class SubstanceService(object):
     def __init__(self):
         self.cache = dict()
 
-    def get_substance_type(self, org, name):
+    def get_extensible_type(self, org, name):
         """
-        Returns a substance_type by name. Can be cached.
+        Returns an extensible_type by name. Can be cached.
         """
+
+        # TODO: This should be in a service only for extensible types, as we'll
+        # require this for ex. containers too.
         if (org.id, name) not in self.cache:
-            substance_type = ExtensibleType.objects.prefetch_related('property_types').get(organization=org, name=name)
+            substance_type = ExtensibleType.objects.prefetch_related(
+                'property_types').get(plugin__organization=org, name=name)
             self.cache[(org.id, name)] = substance_type
         return self.cache[(org.id, name)]
 
     @transaction.atomic
     def copy(self, parent, new_name=None, overridden_properties=None):
         """
-        Copies the parent, giving it a new_name. If new_name is not supplied it will get a unique name
-        based on the name of the parent.
+        Copies the parent, giving it a new_name. If new_name is not supplied it
+        will get a unique name based on the name of the parent.
         """
         overridden_properties = overridden_properties or dict()
 
         if not new_name:
             new_name = "{}:{}".format(parent.name, uuid4())
 
-        child = Substance(name=new_name, organization=parent.organization, extensible_type=parent.extensible_type)
+        child = Substance(
+            name=new_name,
+            organization=parent.organization,
+            extensible_type=parent.extensible_type)
         child.depth = parent.depth + 1
         child.save()
```

Comment:
In this setup, I would argue that django has the repository layer. One can create objects via that layer without saving them. However, when calling the service, I believe that it should actually save the objects, because it's supposed to be used by high level code (the REST API and the plugin API).

Reply:
Ok!